usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateStopTooFarFromTripShapeNoticeTest.java

```diff
@@ -0,0 +1,171 @@
+        when(pt4.getShapeId()).thenReturn(shapeId);
+        when(pt5.getShapeId()).thenReturn(shapeId);
+
+        when(pt1.getShapePtSequence()).thenReturn(1);
+        when(pt2.getShapePtSequence()).thenReturn(2);
+        when(pt3.getShapePtSequence()).thenReturn(3);
+        when(pt4.getShapePtSequence()).thenReturn(4);
+        when(pt5.getShapePtSequence()).thenReturn(5);
+
+        when(pt1.getShapePtLat()).thenReturn(28.05724310653972f);
+        when(pt1.getShapePtLon()).thenReturn(-82.41350776611507f);
+        when(pt2.getShapePtLat()).thenReturn(28.05746701492806f);
+        when(pt2.getShapePtLon()).thenReturn(-82.41493135129478f);
+        when(pt3.getShapePtLat()).thenReturn(28.05800068503469f);
+        when(pt3.getShapePtLon()).thenReturn(-82.4159394137605f);
+        when(pt4.getShapePtLat()).thenReturn(28.05808869825447f);
+        when(pt4.getShapePtLon()).thenReturn(-82.41648754043338f);
+        when(pt5.getShapePtLat()).thenReturn(28.05809979887893f);
+        when(pt5.getShapePtLon()).thenReturn(-82.41773971025437f);
+
+        // trips.txt
+        final String tripId = "trip1";
+        final Trip trip = mock(Trip.class);
+        when(trip.getTripId()).thenReturn(tripId);
+        when(trip.getShapeId()).thenReturn(shapeId);
+
+        // Map containing StopTime entities. Entities are mapped on keys from GTFS file stop_times.txt:
+        // - trip_id
+        // - stop_sequence
+        final Map<String, TreeMap<Integer, StopTime>> stopTimeCollection = new HashMap<>(1);
+        final TreeMap<Integer, StopTime> stopTimes = new TreeMap<>(Map.of(1, stopTime1, 2, stopTime2));
+        stopTimeCollection.put(tripId, stopTimes);
+
+        // Map containing Stop entities. Entities are keyed on GTFS stops.txt stop_id
+        final Map<String, LocationBase> stopPerId = new HashMap<>(Map.of(stopId1, stop1, stopId2, stop2));
+
+        // Entities are keyed on shape_pt_sequence of GTFS file shapes.txt
+        SortedMap<Integer, ShapePoint> points = new TreeMap<>(Map.of(1, pt1, 2, pt2, 3, pt3, 4, pt4, 5, pt5));
+
+        final GtfsDataRepository mockDataRepo = mock(GtfsDataRepository.class);
+        when(mockDataRepo.getStopAll()).thenReturn(stopPerId);
+        when(mockDataRepo.getStopTimeAll()).thenReturn(stopTimeCollection);
+        when(mockDataRepo.getTripById(tripId)).thenReturn(trip);
+        when(mockDataRepo.getShapeById(shapeId)).thenReturn(points);
+        final ValidationResultRepository mockResultRepo = mock(ValidationResultRepository.class);
+        final Logger mockLogger = mock(Logger.class);
+        final GeospatialUtils mockGeoUtil = mock(GeospatialUtils.class);
+
+        final ValidateStopTooFarFromTripShape underTest =
+                new ValidateStopTooFarFromTripShape(mockDataRepo, mockResultRepo,
+                        mockGeoUtil, mockLogger);
+
+        underTest.execute();
+        verify(mockLogger, times(1)).info("Validating rule 'E047 - Stop too far from trip shape'");
```

Comment:
Per discussion with @lionel-nj - just test number of calls to utility method here. Don't need to mock/test spatial calculations.